### PR TITLE
chore: aws infrastructure (docker + deploy workflow)

### PR DESCRIPTION
**Pull request by ntampakas**
## Explanation
This PR dockerizes crypt-keeper-extension elements in order to deploy them against Amazon EC2s using CodeDeploy service.
Also, a new workflow was created + aux scripts.

## More Information
Changes:
- Dockerfile(s)
- docker compose
- deploy workflow
- aux workflow scripts
- codedeploy related manifests + scripts 

## Endpoint
https://ckdemo.appliedzkp.org

### CODE REVIEW

**Review by 0xisk — APPROVED** [review]
Thanks much @ntampakas for your help on this. :ok_hand: 
It's approved, and will leave it for @0xmad to review it as well, and if he has comments we will resolve it as well. 

Do you think @0xmad should we  move `Dockerfiles_demo` and `Dockerfile_merkle` each one to each package?

File: Dockerfile_demo
Comment (0xmad) [inline Dockerfile_demo:1]: Let's move it to demo package

File: Dockerfile_merkle
Comment (0xmad) [inline Dockerfile_merkle:1]: Let's move it to merkle package

File: .github/scripts/build.sh [hunk 1]
```diff
@@ -0,0 +1,20 @@
+#!/bin/bash
+set -ex
+
+build=$1
+
+[ $build = "enable" ] || exit 0
+
+aws ecr get-login-password --region eu-central-1 | docker login --username AWS --password-stdin 490752553772.dkr.ecr.eu-central-1.amazonaws.com
+
+cp .env.example .env
```
[the remark this answers is not in the document]
Comment (ntampakas) [reply .github/scripts/build.sh:10]: Hey! What exactly variable are you talking about? In the command mentioned above, there isn' t any sensitive data. The authentication is becoming possible, using shorted lived credentials (with the appropriate IAM policy)

[the remark this answers is not in the document]
Comment (ntampakas) [reply .github/scripts/build.sh:10]: Hey!
Not sure if it helps to inject GH secrets to EC2s. There was a discussion on discord to use AWS secrets manager, but AFAICS there not sensitive data.

[the remark this answers is not in the document]
Comment (0xmad) [reply .github/scripts/build.sh:10]: Oh, I think we can keep it for now. Demo and merkle don't use any secrets or env variables.

**Review by 0xmad — APPROVED** [review]
@ntampakas thanks!

**Pull request merged**